src/main/java/edu/kit/hci/soli/controller/OpeningHoursController.java

```diff
@@ -0,0 +1,64 @@
+package edu.kit.hci.soli.controller;
+
+import edu.kit.hci.soli.domain.Room;
+import edu.kit.hci.soli.dto.KnownError;
+import edu.kit.hci.soli.dto.LayoutParams;
+import edu.kit.hci.soli.dto.form.SaveOpeningHoursForm;
+import edu.kit.hci.soli.service.RoomService;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+@Controller("/admin/opening-hours")
+public class OpeningHoursController {
+    private final RoomService roomService;
+
+    public OpeningHoursController(RoomService roomService) {
+        this.roomService = roomService;
+    }
+
+    @PutMapping("/admin/opening-hours/{roomId}/save")
```

Comment:
Since this uses the `PUT` method, I think it'd be cleaner (and more REST-like) to align this path with that of the `GET` request